hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionStatusGrid.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.management.actionhistory;
+
+import java.util.Map;
+
+import org.eclipse.hawkbit.repository.model.Action;
+import org.eclipse.hawkbit.ui.common.grid.AbstractGrid;
+import org.eclipse.hawkbit.ui.customrenderers.renderers.HtmlLabelRenderer;
+import org.eclipse.hawkbit.ui.management.actionhistory.ActionHistoryGrid.LabelConfig;
+import org.eclipse.hawkbit.ui.rollout.StatusFontIcon;
+import org.eclipse.hawkbit.ui.utils.SPUIDefinitions;
+import org.eclipse.hawkbit.ui.utils.UIComponentIdProvider;
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+import org.vaadin.addons.lazyquerycontainer.BeanQueryFactory;
+import org.vaadin.addons.lazyquerycontainer.LazyQueryContainer;
+import org.vaadin.addons.lazyquerycontainer.LazyQueryDefinition;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+
+import com.google.common.collect.Maps;
+
+/**
+ * This grid presents the action states for a selected action.
+ */
+public class ActionStatusGrid extends AbstractGrid<LazyQueryContainer> {
+    private static final long serialVersionUID = 1L;
+
+    private static final String[] leftAlignedColumns = new String[] { ProxyActionStatus.PXY_AS_CREATED_AT };
+
+    private static final String[] centerAlignedColumns = new String[] { ProxyActionStatus.PXY_AS_STATUS };
+
+    private final AlignCellStyleGenerator alignGenerator;
+    private final ModifiedTimeTooltipGenerator modTimetooltipGenerator;
+
+    private final Map<Action.Status, StatusFontIcon> states;
+
+    private final BeanQueryFactory<ActionStatusBeanQuery> targetQF = new BeanQueryFactory<>(
+            ActionStatusBeanQuery.class);
+
+    /**
+     * Constructor.
+     *
+     * @param i18n
+     * @param eventBus
+     */
+    protected ActionStatusGrid(final VaadinMessageSource i18n, final UIEventBus eventBus) {
+        super(i18n, eventBus, null);
+
+        setSingleSelectionSupport(new SingleSelectionSupport(true));
+        setDetailsSupport(new DetailsSupport());
+
+        final LabelConfig conf = new ActionHistoryGrid.LabelConfig();
+        states = conf.createStatusLabelConfig(i18n, UIComponentIdProvider.ACTION_STATUS_GRID_STATUS_LABEL_ID);
+        alignGenerator = new AlignCellStyleGenerator(leftAlignedColumns, centerAlignedColumns, null);
+        modTimetooltipGenerator = new ModifiedTimeTooltipGenerator(ProxyActionStatus.PXY_AS_CREATED_AT);
+
+        init();
```

Comment:
Constructor should not call overridables see: http://www.javapractices.com/topic/TopicAction.do?Id=215